### PR TITLE
feat: display contact search results from kalium

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- display search results coming from the back-end
- handle loading and error state
- handle the ui logic for results having less than 4 items
- handle adding the contacts to the group

search_result

#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt [hunk 1]
```diff
@@ -165,27 +167,43 @@ class UseCaseModule {
...
     fun providesSendTextMessageUseCase(
         @KaliumCoreLogic coreLogic: CoreLogic,
         @CurrentSession currentSession: AuthSession
     ): SendTextMessageUseCase =
-        coreLogic.getSessionScope(currentSession).messages.sendTextMessage
+        coreLogic.getSessionScope(currentSession.userId).messages.sendTextMessage
+
+    @ViewModelScoped
+    @Provides
+    fun providesSearchKnownUsersUseCase(
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:190]: What do you think of calling knownUsers as connectedUsers?

Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:190]: do you mean changing the name only on the reloaded side or also Kalium ?

**Review by ohassine — CHANGES_REQUESTED** [review]
Great effort! But that's too many changes at once, big PRs are hard to review and may lead to error prone. please divide this PR into smaller ones

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt (hunk 1 above)
Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:190]: I would say both, as we should keep it as consistent as possible between client and core

Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt:190]: there was lots of disscussion ongoing about the naming and this is something we agreed so far on Kalium side, so I am going to keep it for now

File: app/src/main/kotlin/com/wire/android/ui/common/WireLabelledCheckbox.kt [hunk 2]
```diff
@@ -52,3 +53,17 @@ fun WireLabelledCheckbox(
         )
     }
 }
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun WiredCheckbox(
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/common/WireLabelledCheckbox.kt:59]:
```suggestion
fun WireCheckbox(
```

File: app/src/main/kotlin/com/wire/android/ui/common/button/WireCheckBoxDefaults.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.state.ToggleableState
+import com.wire.android.ui.theme.wireColorScheme
+
+@Composable
+fun wireCheckBoxColors() = wireCheckBoxColors(
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/common/button/WireCheckBoxDefaults.kt:15]: Niceee 👏

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/common/GroupButton.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import com.wire.android.ui.common.textfield.WirePrimaryButton
+import com.wire.android.ui.theme.wireDimensions
+
+@Composable
+fun GroupButton(groupSize: Int) {
+    Row(
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .height(82.dp)
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/common/GroupButton.kt:33]: I feel like I have been seeing this magic `82.dp` number way too many times. Can we extract it to a wire dimension value?

Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/common/GroupButton.kt:33]: if it has nothing to do with the send button of the Message Composer, can we use `spacing80x` instead?

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsScreen.kt [hunk 5]
```diff
@@ -46,67 +49,93 @@ fun ContactsScreenContent(
...
-        )
+    RowItemTemplate(
+        leadingIcon = {
+            Row {
+                WiredCheckbox(checked = belongsToGroup, onCheckedChange = { if (it) addToGroup() else removeFromGroup() })
+                UserProfileAvatar(
+                    status = userStatus
+                )
+            }
+
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsScreen.kt:98]:
```suggestion
```

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsScreen.kt [hunk 6]
```diff
@@ -46,67 +49,93 @@ fun ContactsScreenContent(
...
+        actions = {
+            Box(
+                modifier = Modifier
+                    .wrapContentWidth()
+                    .padding(end = 8.dp)
+            ) {
+                ArrowRightIcon(Modifier.align(Alignment.TopEnd))
+            }
+
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsScreen.kt:115]:
```suggestion
```

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsState.kt [hunk 7]
```diff
@@ -11,13 +11,11 @@ data class Contact(
...
-data class ExternalContact(
-    val id: Int,
-    val name: String,
-    val userStatus: UserStatus = UserStatus.AVAILABLE,
-    val avatarUrl: String = "",
-    val label: String = "",
-)
+fun PublicUser.toContact() =
+    Contact(
+        id = 1,
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsState.kt:18]: Typo? or why are we hardcoding this?

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/HighLightSubtTitle.kt [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+import androidx.compose.ui.text.withStyle
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.ui.theme.wireTypography
+import com.wire.android.util.MatchQueryResult
+import com.wire.android.util.QueryMatchExtractor
+import kotlinx.coroutines.launch
+
+
+@Composable
+fun HighLightSubTitle(
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/HighLightSubtTitle.kt:23]:
```suggestion
fun HighlightSubtitle(
```

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/HighLightName.kt [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.withStyle
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.ui.theme.wireTypography
+import com.wire.android.util.MatchQueryResult
+import com.wire.android.util.QueryMatchExtractor
+import kotlinx.coroutines.launch
+
+@Composable
+fun HighLightName(
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/HighLightName.kt:22]:
```suggestion
fun HighlightName(
```

**Review by gongracr — COMMENTED** [review]
Just small comments. Awesome work!!!! 🛎️

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsState.kt (hunk 7 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/contacts/ContactsState.kt:18]: typo <3 thanks

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/InternalContactSearchResultItem.kt [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+                )
+            }
+        },
+        title = {
+            HighlightName(
+                name = name,
+                searchQuery = searchQuery
+            )
+        },
+        subTitle = {
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/InternalContactSearchResultItem.kt:53]:
```suggestion
        subtitle = {
```

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleViewModel.kt [hunk 11]
```diff
@@ -1,80 +1,109 @@
...
-                        )
-                    )
-                }
-            }
+class SearchPeopleViewModel @Inject constructor(
+    private val searchKnownUsers: SearchKnownUsersUseCase,
+    private val searchPublicUsers: SearchUserDirectoryUseCase
+) : ViewModel() {
+    private companion object {
+        const val HARDCODED_TEST_DOMAIN = "staging.zinfra.io"
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleViewModel.kt:31]: I will add a TODO annotation here to remember mapping it out with the given backend configuration later on

**Review by gongracr — APPROVED** [review]
I would assume we still want to polish a bit the UI (mocked data, and animations) but all in all fantastic work 👏 🎩 🔍 
The time to fully start a conversation is almost here 🐲 🔈

**Review by gongracr — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleViewModel.kt (hunk 11 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleViewModel.kt:32]: done

**Pull request merged**